source/dubregistry/dbcontroller.d

```diff
@@ -177,6 +177,35 @@ class DbController {
 		m_packages.update(["name": packname], ["$set": ["repository": repo]]);
 	}
 
+	void setPackageLogo(string packname, bdata_t png)
+	{
+		import std.digest.md : md5Of;
+
+		if (png.length)
+		{
+			m_packages.update(["name": packname], ["$set": [
+				"logo": BsonBinData(BsonBinData.Type.generic, png),
+				"logoHash": BsonBinData(BsonBinData.Type.md5, cast(bdata_t)md5Of(png)[].idup)
+			]]);
+		}
+		else
+			m_packages.update(["name": packname], ["$unset": ["logo": 1, "logoHash": 1]]);
+	}
+
+	bdata_t getPackageLogo(string packname, out bdata_t rev)
```

Comment:
This a question of design/personal preferences, but I typically return tuples in such cases. Hence out of interest, why did you go with `out`?

Reply:
I go `struct` or `out`, but never tuple because it requires me to import something rather large. Selective imports don't help with compilation time in most of phobos.

Reply:
`std.typecons` is already imported in other modules (and even in vibe.d), so there's no extra import cost.
Also FWIW we reduced the total import time of all Phobos modules to < 0.5s with 2.079. On my machine it's even less than 0.2s now

See: https://dlang.org/changelog/2.079.0.html#std-experimental-all